### PR TITLE
Add helper tools in Makefile

### PR DESCRIPTION
**Pull request by jedevc**
This PR adds a `Makefile` helper tool with a collection of useful utilities:

- `make build` to build the site into the `_site` directory
  - `make clean` to tidy up the artifacts from `make build`
- `make serve` to serve the site in a docker container (without messing up the `_site` directory, and making it owned by root :angry:, which is how I had it setup before)
- `make fmt` to use [prettier](https://prettier.io/) to autoformat SCSS and JS code, to prevent mixed tabs/spaces and to enforce a consistent style

The second commit is the result of running `make fmt` over the codebase :tada:

I can easily remove the auto-formatter if it's too controversial, but I very much like it :eyes: Ideally, it could also be integrated into a github action to cause a failure if `make fmt` hasn't been run.

### CODE REVIEW

File: .prettierignore [hunk 1]
```diff
@@ -0,0 +1,5 @@
+*
+!**/
+!*.scss
+!*.js
+js/redirect.js
```
Comment (LMBishop) [inline .prettierignore:5]:
```suggestion
**/js/redirect.js
```

Or `_site` instead, otherwise `prettier --check` will fail if the website was built

**Review by LMBishop — COMMENTED** [review]
This supersedes #285. Do you think you could update the code style section of the [contributing guide](https://github.com/CSSUoB/cssuob.github.io/blob/master/CONTRIBUTING.md) as well? If not, then don't worry.

File: .prettierignore (hunk 1 above)
Comment (jedevc) [reply .prettierignore]: Ooh nice catch. Have added `_site`, no reason to check the built site.

**Comment by jedevc** [comment]
> Do you think you could update the code style section of the [contributing guide](https://github.com/CSSUoB/cssuob.github.io/blob/master/CONTRIBUTING.md) as well? If not, then don't worry.

Have done :)

**Review by LMBishop — APPROVED** [review]
Awesome :)

**Review by MattyTheHacker — APPROVED** [review]
looks good, I'll close my cleanup PR when this is merged

**Comment by LMBishop** [comment]
@jedevc one final thing (sorry), could you rebase your branch and discard b8e1313, then run `make fmt` again? This is so it doesn't undo your change in #317.

**Comment by jedevc** [comment]
Woops, conflict resolved.

**Pull request merged**